docs/addons/nats/tls/index.md

```diff
@@ -0,0 +1,783 @@
+
+## Before You Begin
+
+- At first, you need to have a Kubernetes cluster, and the `kubectl` command-line tool must be configured to communicate with your cluster.
+- Install Stash Enterprise in your cluster following the steps [here](/docs/setup/install/enterprise.md).
+- Install cert-manager in your cluster following the instruction [here](https://cert-manager.io/docs/installation/)
+- If you are not familiar with how Stash backup and restore NATS streams, please check the following guide [here](/docs/addons/nats/overview/index.md).
+
+You have to be familiar with following custom resources:
+
+- [AppBinding](/docs/concepts/crds/appbinding.md)
+- [Function](/docs/concepts/crds/function.md)
+- [Task](/docs/concepts/crds/task.md)
+- [BackupConfiguration](/docs/concepts/crds/backupconfiguration.md)
+- [BackupSession](/docs/concepts/crds/backupsession.md)
+- [RestoreSession](/docs/concepts/crds/restoresession.md)
+
+To keep things isolated, we are going to use a separate namespace called `demo` throughout this tutorial. Create `demo` namespace if you haven't created already.
+
+```bash
+$ kubectl create ns demo
+namespace/demo created
+```
+
+> Note: YAML files used in this tutorial are stored [here](https://github.com/stashed/docs/tree/{{< param "info.version" >}}/docs/addons/nats/tls/examples).
+
+## Prepare NATS
+
+In this section, we are going to deploy a TLS secured NATS cluster. Then, we are going to create a stream and publish some messages into it.
+
+
+### Create Certificate
+To issue a CA certificate we have to create a `ClusterIssuer`. Below is the YAML of `ClusterIssuer` object we are going to create.
+```yaml
+apiVersion: cert-manager.io/v1
+kind: ClusterIssuer
+metadata:
+  name: selfsigning
+  namespace: demo
+spec:
+  selfSigned: {}
+```
+Let's create the `ClusterIssuer` we have shown above,
+```bash
+$ kubectl apply -f https://github.com/stashed/docs/tree/{{< param "info.version" >}}/docs/addons/nats/tls/examples/clusterissuer.yaml
+clusterissuer.cert-manager.io/selfsigning created
+```
+
+Now we are going to create a CA certificate issued by the the `ClusterIssuer`. Below is the YAML of `Certificate` object we are going to create.
+```yaml
+apiVersion: cert-manager.io/v1
+kind: Certificate
+metadata:
+  name: nats-ca
+  namespace: demo
+spec:
+  secretName: nats-ca
+  duration: 8736h # 1 year
+  renewBefore: 240h # 10 days
+  issuerRef:
+    name: selfsigning
+    kind: ClusterIssuer
+  commonName: nats-ca
+  isCA: true
+```
+Let's create the `Certificate` we have shown above,
+```bash
+$ kubectl apply -f https://github.com/stashed/docs/tree/{{< param "info.version" >}}/docs/addons/nats/tls/examples/ca.yaml
+certificate.cert-manager.io/nats-ca created
+```
+
+To issue a TLS certificate we have to create an `Issuer`. Below is the YAML of `Issuer` object we are going to create.
```

Comment:
Cert-manager will automatically create a Secret named specified by `spec.secretName` filed with the desired CA certificate. Let's verify that the Secret has been created successfully,
```bash
kubectl get secret -n demo nats-ca
```
Now, we are going create a `Issuer` with the above CA Secret. We are going to use this `Issuer` to issue server and client certificates for our NATS server. Below is the ....